docker/config/excerpts/kimobjects.py

```diff
@@ -641,6 +622,19 @@ def simulator_potential(self):
                 "kimspec.edn file of {}".format(self.kim_code)
             )
         return self.kimspec["simulator-potential"]
+    
+    @property
+    def run_compatibility(self):
+        """
+        Whether the SM can run against regular Tests (that are designed to run
+        against portable models) as opposed to Tests that are specifically
+        constructed to run against a particular subclass of SMs.
+        """
+        if not self.kimspec:
+            return None
+        else:
+            return self.kimspec["run-compatibility"]
+
```

Comment:
We've added run-compatibility to all SMs and it is now a required key for SMs, so we've updated the logic both here and the pipeline to reflect this. In fact, we have our first special-purpose test result here: https://openkim.org/id/TE_102563389665_000-and-SM_672022050407_000-tr

A stale version would crash, which is fine because it's a required key now, but it's a good point -- I should make the error message more informative. Right now I believe they would get a message saying

                    "Runner %s lists 'matching-models ['standard-models'] "
                "but subject %s lists 'run-compatibility' 'special-purpose-models'"

Which is misleading

Reply:
Returning to this, I don't think a change is necessary -- it will not give the misleading error as I thought, it gives a KeyError for not finding 'run-compatibility' which seems clear enough to me.